### DISCUSSION TITLE
How can we change the sample model (checkpoint) when creating samples during LoRA training?

### DISCUSSION THREAD

**Discussion by gituser123456789000** [bmaltais/kohya_ss#3026] [Q&A]
I assume the samples seem pretty clearly to be using the base sdxl model, which nobody that knows what they're doing would still be using.. they would be using some fine-tuned checkpoint. 

As a result of using the base model, the results are often poor looking and little indication of how results will look when using a LoRA after it's created. So how do we change what checkpoint is being used when creating sample images during LoRA training?

I looked through kohya;s files as best I know how and I didn't see the base model sdxl file. No folders seem to be big enough to be holding an SDXL checkpoint as far as I see. So I don't even know how kohya is able to create samples to begin with, without the model somewhere in the program.

If the model is there somewhere, I'd assume it should be pretty simple to delete the base model and put a different, more current, fine-tuned model in its place and just rename it to the base model's name that works in the code.

Does anyone already know or can figure out a solution for this?

And for a feature suggestion, .....

hmmm... Now that I think about it, I suppose it uses the model you set as the model you're training the LoRA on to create the samples.

But I assume most people train the LoRA on the base model, so that it works on all/most fine-tunes, even though they don't use the base model when later using the LoRA..

So feature suggestion would be to add an option in the Sample section to set a different model/checkpoint to use when creating samples during LoRA training. That way users can train on the base model, but get actually usable sample results with a model/checkpoint they'd actually be using the LoRA with.

**Comment by gituser123456789000** [comment]
Or is there some code to add to the sample prompts to change the sampler checkpoint directory?

Like I know to put an ending like --w 1024 --h 1024 --l 2.0 --s 40
To set dimensions, cfg I assume is the 2.0, and sample steps...
Is there some addition code that could be added like, ---checkpoint D\models\finetunedmodel.safetensors?

**Comment by Sylsatra** [comment]
Umm the sample actually uses the training model, you cant find it because you didnt save it every epoch.

**Comment by jt-michels** [comment]
It uses whatever model you're training... So the option is there  when you start your training.  It would be counterintuitive to run inference on a separate model during training.
 
And also, (depending on use case—there is leeway) you should be training on the model you intend to use the LoRA with.  Weights shift and change the more the model is finetuned and/or merged so if you train a LoRA on [base] and then  put that LoRA on a finetuned models, the LoRA is still giving instructions to adjust the weights as if you are using the [base].  Eventually the weights will shift enough for the inference to be unrecognizable with the [base] LoRA (take Pony or Illustrious for example, these are SDXL checkpoints that have been trained significantly enough for most of the weights and biases to be unrecognizable). So if you don't plan on using SDXL 1.0 Base (which PS is pretty versatile and I think people still use for general use case; I keep a copy to use ad hoc if needed...it's also easier to train on for newer trainers), you should be using your model of choice.
